src/execution_impl.rs

```diff
@@ -72,6 +72,55 @@ fn create_instance(limit: u64, module: &[u8], interface: &dyn Interface) -> Resu
     Ok(Instance::new(&module, &resolver)?)
 }
 
+/// Closure for the execution allowing us to handle a gas error
+/// used by the [exec] function.
+fn internal_execution(
```

Comment:
let's document parameters and return type

Reply:
Done.